lib/modules/manager/asdf/extract.spec.ts

```diff
@@ -0,0 +1,160 @@
+import { extractPackageFile } from '.';
+
+describe('modules/manager/asdf/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('returns a result', () => {
+      const res = extractPackageFile('nodejs 16.16.0\n');
+      expect(res).toEqual({
+        deps: [
+          {
+            currentValue: '16.16.0',
+            datasource: 'github-tags',
+            depName: 'node',
+            packageName: 'nodejs/node',
+            versioning: 'node',
+          },
+        ],
+      });
+    });
+
+    it('provides skipReason for lines with unsupported tooling', () => {
+      const res = extractPackageFile('unsupported 1.22.5\n');
+      expect(res).toEqual({
+        deps: [
+          {
+            depName: 'unsupported',
+            skipReason: 'unsupported-datasource',
+          },
+        ],
+      });
+    });
+
+    it('only captures the first version', () => {
+      const res = extractPackageFile('nodejs 16.16.0 16.15.1\n');
+      expect(res).toEqual({
+        deps: [
+          {
+            currentValue: '16.16.0',
+            datasource: 'github-tags',
+            depName: 'node',
+            packageName: 'nodejs/node',
+            versioning: 'node',
+          },
+        ],
+      });
+    });
+
+    it('provides skipReason for supported tooling with ref: versioning', () => {
+      const res = extractPackageFile('nodejs ref:234abc4\n');
+      expect(res).toEqual({
+        deps: [
+          {
+            currentValue: 'ref:234abc4',
+            depName: 'node',
+            skipReason: 'unsupported-version',
+          },
+        ],
+      });
+    });
+
+    it('provides skipReason for supported tooling with path: versioning', () => {
+      const res = extractPackageFile('nodejs path:/path/to/tooling\n');
+      expect(res).toEqual({
+        deps: [
+          {
+            currentValue: 'path:/path/to/tooling',
+            depName: 'node',
+            skipReason: 'unsupported-version',
+          },
+        ],
+      });
+    });
+
+    it('provides skipReason for supported tooling with system versioning', () => {
+      const res = extractPackageFile('nodejs system\n');
+      expect(res).toEqual({
+        deps: [
+          {
+            currentValue: 'system',
+            depName: 'node',
+            skipReason: 'unsupported-version',
+          },
+        ],
+      });
+    });
+
+    it('can handle multiple tools in one file', () => {
+      const res = extractPackageFile('nodejs 16.16.0\nyarn 1.2.3');
```

Comment:
```suggestion
      const res = extractPackageFile('nodejs 16.16.0\ndummy 1.2.3');
```
we should use a dummy tool, as yarn will be updateable soon, so we don't need to change that test 😉

Reply:
ah - thanks - I wasn't aware of that and that does make sense